terraform/modules/alb-listener-ip-restriction-rules/main.tf

```diff
@@ -1,18 +1,5 @@
-locals {
-  # Note: Needed due to an AWS quota restriction (limit of 5 conditions per rule)
-  # In future we may need to request a service quota increase of
-  # "Rules per Application Load Balancer" from AWS.
-  rules = setproduct(var.restricted_path_patterns, var.fully_trusted_source_ips)
```

Comment:
You note in the commit message using `count` is inappropriate as one rule can support multiple path patterns and  source IPs. That's true, but there is a limitation to this that the Terraform docs don't capture - and isn't caught by a `terraform plan`.

We used `count` here to handle an AWS limit: if I recall correctly, AWS sets a limit of 5 elements in the values list for `path_pattern` and `source_ip`.

I think this constraint is documented as  "Match evaluations per rule: 5" in the [AWS docs](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-limits.html).

We hit this constraint as we have more than 5 trusted IPs.

Reply:
cool, I overlook that fact. I think if this is the case,we might be better off using a WAF.